src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/String/KeyAnalyzer.cs

```diff
@@ -263,25 +245,34 @@ public AnalysisResults(bool ignoreCase, bool allAsciiIfIgnoreCase, int hashIndex
             public bool RightJustifiedSubstring => HashIndex < 0;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static ReadOnlySpan<char> Slicer(this string s, int index, int count) => s.AsSpan((index >= 0 ? index : s.Length + index), count);
```

Comment:
Could roll in the [logic for non-substring from line 116](https://github.com/dotnet/runtime/pull/88516/files#diff-8dfef3756fbf6ca0aab7668e6abf54ae6c416c60cb1b3bd86455204942f7dbefR116) as this gets inlined, might even make a tiny bit better JIT.